_posts/2021-03-11-Playing-along-with-NFTables.md

```diff
@@ -0,0 +1,294 @@
+```
+
+Now our wanted_tcp_ports list will allow port 22 and 443.
+This is of course often more useful if we use it with IP addresses.
+
+```
+    set fail2ban_hosts {
+        type ipv4_addr; flags interval;
+        elements = {
+           192.168.0.0/24
+        }       
+    }
+```
+
+Let us append some elements to that set too.
+
+```
+$ nft add element inet filter fail2ban_hosts \{ 192.168.254.255, 192.168.253.0/24 \}
+$ nft list ruleset
+```
+
+... and we get ... 
+
+```
+        set fail2ban_hosts {
+                type ipv4_addr
+                flags interval
+                elements = { 192.168.0.0/24, 192.168.253.0/24,
+                             192.168.254.255 }
+        }
+```
+
+Now we could change fail2ban to append elements to the set instead of creating a new rule for each new machine it wants to block. Fewer rules. Faster processing.
+
+But with reloading the firewall we dropped port 443 from the port list again. Oops.
+Though ... if you are happy with the rules. You can just run
+
+```
+$ nft list ruleset > nftables.conf
+```
+
+When you are using all the sets instead of the variables, all your firewall rules will still look nice.
+
+Our complete firewall looks like
+
+```
+table inet filter {
+        set wanted_tcp_ports {
+                type inet_service
+                flags interval
+                elements = { 22, 443 }
+        }
+
+        set fail2ban_hosts {
+                type ipv4_addr
+                flags interval
+                elements = { 192.168.0.0/24, 192.168.253.0/24,
+                             192.168.254.255 }
+        }
+
+        chain base_checks {
+                ct state { established, related } accept
+                ct state invalid drop
+        }
+
+        chain input {
+                type filter hook input priority filter; policy drop;
+                iif "lo" accept
+                jump base_checks
+                ip6 nexthdr ipv6-icmp icmpv6 type { destination-unreachable, packet-too-big, time-exceeded, parameter-problem, echo-request, echo-reply, mld-listener-query, mld-listener-report, mld-listener-done, nd-router-solicit, nd-router-advert, nd-neighbor-solicit, nd-neighbor-advert, ind-neighbor-solicit, ind-neighbor-advert, mld2-listener-report } accept
+                ip protocol icmp icmp type { echo-reply, destination-unreachable, echo-request, router-advertisement, router-solicitation, time-exceeded, parameter-problem } accept
+                ip protocol igmp accept
+                tcp dport @wanted_tcp_ports accept
+                counter packets 12 bytes 828 log prefix "[nftables] input reject " reject
+        }
+
+        chain forward {
+                type filter hook forward priority filter; policy accept;
+        }
+
+        chain output {
+                type filter hook output priority filter; policy accept;
+        }
+}
+```
+
+For more see the [nftables wiki](https://wiki.nftables.org/wiki-nftables/index.php/Main_Page)
+
+
+djaksjdk
```

Comment:
I guess we don't need the last line ;-)

(also already fixed on https://progress.opensuse.org/news/112)